ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -238,23 +249,54 @@ private boolean isSourceFileMismatch(FileSystem sourceFs, ReplChangeManager.File
     return false;
   }
 
+  private UserGroupInformation getProxyUser() throws LoginException, IOException {
+    if (copyAsUser == null) {
+      return null;
+    }
+    UserGroupInformation proxyUser = null;
+    int currentRetry = 0;
+    while (currentRetry <= MAX_IO_RETRY) {
+      try {
+        UserGroupInformation ugi = Utils.getUGI();
+        String currentUser = ugi.getShortUserName();
+        if (!currentUser.equals(copyAsUser)) {
+          proxyUser = UserGroupInformation.createProxyUser(
+                  copyAsUser, UserGroupInformation.getLoginUser());
+        }
+        return proxyUser;
+      } catch (IOException e) {
+        currentRetry++;
+        if (currentRetry <= MAX_IO_RETRY) {
+          LOG.warn("Unable to get UGI info", e);
```

Comment:
Can log the retry number in the log msg.

Reply:
ok